### ISSUE TITLE
YARD:: Parser:: Ruby:: Method Definition Node#parameters

### ISSUE THREAD

**Issue by MSP-Greg** [lsegal/yard#1015]
See

https://github.com/lsegal/yard/blob/master/lib/yard/parser/ruby/ast_node.rb#L472

While investigating the issue with the -1 default value, came across the following code:

``` ruby
def parameters(include_block_param = true)
  params = self[1 + index_adjust]
  params = params[0] if params.type == :paren
  include_block_param ? params : params[0...-1]
end
```

last line in method, `params[0...-1]` removes an item from _both_ ends of the array.

Hence, it seems that the line should be --

``` ruby
include_block_param ? params : params[0..-2]
```

Maybe I've been staring at code for too long...
## Environment details:
- OS: [Windows]
- Ruby version: [2.3.2]
- YARD version: [0.9.5]
- Relevant software dependency/versions:
  - [NA]

I have read the [Contributing Guide](https://github.com/lsegal/yard/blob/master/CONTRIBUTING.md).

**Comment by MSP-Greg** [comment]
Sorry, forgot to mention that
```ruby
include_block_param ? params : params[0..-2]
```

passes RSpec using 2.3.2 & 2.2.4.  Haven't investigated further.

**Comment by lsegal** [comment]
If changing this line passes tests, there are probably no tests for this actual functionality. It would be worth adding tests (including ones that address the issue from #894) to see if your fix resolves them.

**Comment by MSP-Greg** [comment]
To clarify, the fix for percent arrays does not fix #894.  Still working on that, maybe I'll get back to it today.

Re the tests, you'd probably zero right in on this one, but I'll have a look.

I did a bit more work yesterday checking into #894, as I'd like to find some more failure cases (one's complement `~` also fails), but that's as far as I've gotten.  Speaking of blocks, I haven't checked if they are affected by #894.  One would think not, but then again...

**Comment by lsegal** [comment]
Closing out some old issues. Given the significant age of this, it would be best to re-open with new context if it's still relevant.
